app/src/main/java/com/lagradost/quicknovel/providers/RanobesProvider.kt

```diff
@@ -0,0 +1,264 @@
+    override val mainUrl = "https://ranobes.net"
+    override val hasMainPage = true
+
+    override val iconId = R.drawable.icon_ranobes
+
+    override val iconBackgroundId = R.color.white
+
+    override val tags = listOf(
+        Pair("Action", "Action"),
+        Pair("Adult", "Adult"),
+        Pair("Adventure", "Adventure"),
+        Pair("Comedy", "Comedy"),
+        Pair("Drama", "Drama"),
+        Pair("Ecchi", "Ecchi"),
+        Pair("Fantasy", "Fantasy"),
+        Pair("Game", "Game"),
+        Pair("Gender Bender", "Gender%20Bender"),
+        Pair("Harem", "Harem"),
+        Pair("Josei", "Josei"),
+        Pair("Historical", "Historical"),
+        Pair("Horror", "Horror"),
+        Pair("Martial Arts", "Martial%20Arts"),
+        Pair("Mature", "Mature"),
+        Pair("Mecha", "Mecha"),
+        Pair("Mystery", "Mystery"),
+        Pair("Psychological", "Psychological"),
+        Pair("Romance", "Romance"),
+        Pair("School Life", "School%20Life"),
+        Pair("Sci-fi", "Sci-fi"),
+        Pair("Seinen", "Seinen"),
+        Pair("Shoujo", "Shoujo"),
+        Pair("Shounen", "Shounen"),
+        Pair("Slice of Life", "Slice%20of%20Life"),
+        Pair("Shounen Ai", "Shounen%20Ai"),
+        Pair("Sports", "Sports"),
+        Pair("Supernatural", "Supernatural"),
+        Pair("Smut", "Smut"),
+        Pair("Tragedy", "Tragedy"),
+        Pair("Xianxia", "Xianxia"),
+        Pair("Xuanhuan", "Xuanhuan"),
+        Pair("Wuxia", "Wuxia"),
+        Pair("Yaoi", "Yaoi"),
+    )
+
+    override fun loadMainPage(
+        page: Int,
+        mainCategory: String?,
+        orderBy: String?,
+        tag: String?
+    ): HeadMainPageResponse {
+        val url = "$mainUrl/tags/genre/$tag/"
+        val response = khttp.get(url)
+
+        val document = Jsoup.parse(response.text)
+
+        val headers = document.select("div.short-cont")
+        if (headers.size <= 0) return HeadMainPageResponse(url, ArrayList())
+        val returnValue: ArrayList<SearchResponse> = ArrayList()
+        for (h in headers) {
+            val h2 = h.selectFirst("h2.title > a")
+            val cUrl = h2.attr("href")
+
+            val name = h2.text()
+            val posterUrl = h.selectFirst("div.cont.showcont > div > a > figure").attr("style").replace("""background-image: url(""","").replace(");","")
+
+            val latestChap = mainUrl + h.nextElementSibling().selectFirst("div > a").attr("href")
+            returnValue.add(
+                SearchResponse(
+                    name,
+                    cUrl,
+                    fixUrl(posterUrl),
+                    null,
+                    latestChap,
+                    this.name
+                )
+            )
+        }
+        return HeadMainPageResponse(url, returnValue)
+    }
+
+    override fun loadHtml(url: String): String? {
+        val response = khttp.get(url)
+        val document = Jsoup.parse(response.text)
+        Thread.sleep(Random.nextLong(45,90))
```

Comment:
You can override the delay between requests rateLimitTime instead of doing it here

Reply:
I think that it has to stay like that. I will ask Addipe about that. I did make it able to load novels with 1000+ chapters now